src/display/editor/freetext.js

```diff
@@ -537,24 +568,65 @@ class FreeTextEditor extends AnnotationEditor {
       return null;
     }
 
+    if (this.deleted) {
+      return {
+        pageIndex: this.pageIndex,
+        id: this.annotationElementId,
+        deleted: true,
+      };
+    }
+
     const padding = FreeTextEditor._internalPadding * this.parentScale;
     const rect = this.getRect(padding, padding);
-
     const color = AnnotationEditor._colorManager.convert(
       this.isAttachedToDOM
         ? getComputedStyle(this.editorDiv).color
         : this.#color
     );
 
-    return {
+    const serialized = {
       annotationType: AnnotationEditorType.FREETEXT,
       color,
       fontSize: this.#fontSize,
       value: this.#content,
       pageIndex: this.pageIndex,
       rect,
       rotation: this.rotation,
+      id: this.annotationElementId,
     };
+
+    if (this.annotationElementId && !this.#hasElementChanged(serialized)) {
+      return null;
+    }
+
+    return serialized;
+  }
+
+  #hasElementChanged(serialized) {
+    const { value, fontSize, color, rect, pageIndex } = this.#initialData;
+
+    return (
+      serialized.value !== value ||
+      serialized.fontSize !== fontSize ||
+      serialized.rect.some((x, i) => Math.abs(x - rect[i]) >= 1) ||
+      serialized.color.some((c, i) => c !== color[i]) ||
+      serialized.pageIndex !== pageIndex
+    );
+  }
+
+  #cheatInitialRect(delayed = false) {
```

Comment:
Please add a comment explaining what this method does, and when and why we need it, since the name doesn't feel all that clear to me.